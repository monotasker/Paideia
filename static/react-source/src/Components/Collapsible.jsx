import React, { useState } from "react";
import {
  Collapse,
  Card
} from "react-bootstrap";

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faPlus,
  faMinus,
  // faNetworkWired
} from '@fortawesome/free-solid-svg-icons';

const Collapsible = ({linkElement="h3",
                      linkIcon,
                      linkText,
                      children,
                      ...otherProps}) => {
  const [open, setOpen] = useState(false);
  const Tag = linkElement;

  return (
    <Card {...otherProps} >
      <Card.Body>
        <a onClick={() => setOpen(!open)}
          aria-controls="collapse-pane"
          aria-expanded={open}
        >
          <Tag>
            {!!linkIcon &&
              <FontAwesomeIcon icon={linkIcon} fixedWidth />
            }
            {linkText}
            <FontAwesomeIcon icon={open ? faMinus : faPlus} pull="right" />
          </Tag>
        </a>
        <Collapse in={open}>
          <div id="switching-pane">
            {children}
          </div>
        </Collapse>
      </Card.Body>
    </Card>
  )
}

// class Collapsible extends Component {
//   constructor(props, context) {
//     super(props, context);

//     this.state = {
//       open: false,
//     };
//   }

//   static defaultProps = {
//     linkText: null,
//     linkElement: "h3"
//   }

//   render() {
//     const { open } = this.state;
//     const Tag = this.props.linkElement;
//     return(
//       <Card>
//         <Card.Body>
//           <a onClick={() => this.setState({open: !open})}
//             aria-controls="collapse-pane"
//             aria-expanded={open}
//           >
//             <Tag>
//               {!!this.props.linkIcon &&
//                 <FontAwesomeIcon icon={this.props.linkIcon} fixedWidth />
//               }
//               {this.props.linkText}
//               <FontAwesomeIcon icon={open ? faMinus : faPlus} pull="right" />
//             </Tag>
//           </a>
//           <Collapse in={this.state.open}>
//             <div id="switching-pane">
//               {this.props.children}
//             </div>
//           </Collapse>
//         </Card.Body>
//       </Card>
//     )
//   }
// }

export default Collapsible;
