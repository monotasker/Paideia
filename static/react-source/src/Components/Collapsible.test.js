import React from 'react';
import Collapsible from "./Collapsible";
import renderer from "react-test-renderer";

test('Pane opens and collapses when title is clicked', () => {
  const component = renderer.create(
    <Collapsible titleText="My title" titleElement="h4" />
  );
  const title = component.root.findByType('a');

  //starting state
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(title.props["aria-expanded"]).toBe(false);

  // manually trigger event
  title.props.onClick();
  // re-render
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(title.props["aria-expanded"]).toBe(true);

  // manually trigger event
  title.props.onClick();
  // re-render
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(title.props["aria-expanded"]).toBe(false);
});



    // <div class="card">
    //   <div class="card-body">
    //     <a aria-controls="collapse-pane" aria-expanded="false">
    //       <h4>How Do I Switch to a Greek Keyboard?
    //         <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="plus" class="svg-inline--fa fa-plus fa-w-14 fa-pull-right " role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"></path></svg>
    //       </h4>
    //     </a>
    //     <div id="switching-pane" class="collapse">
    //     </div>
    //   </div>
    // </div>
