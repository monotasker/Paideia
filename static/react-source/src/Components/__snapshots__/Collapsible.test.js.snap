// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pane opens and collapses when title is clicked 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <a
      aria-controls="collapse-pane"
      aria-expanded={false}
      onClick={[Function]}
    >
      <h3>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-w-14 fa-pull-right "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </h3>
    </a>
    <div
      aria-expanded={null}
      className="collapse"
      id="switching-pane"
    />
  </div>
</div>
`;

exports[`Pane opens and collapses when title is clicked 2`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <a
      aria-controls="collapse-pane"
      aria-expanded={false}
      onClick={[Function]}
    >
      <h3>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-w-14 fa-pull-right "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </h3>
    </a>
    <div
      aria-expanded={null}
      className="collapse"
      id="switching-pane"
    />
  </div>
</div>
`;
